src/api/api.ts

```diff
@@ -0,0 +1,20 @@
+import { client } from './graphql';
+import { GET_CATEGORIES } from './queries';
+
+export interface Category {
+  id: string;
+  name: string;
+  imageUrl: string;
+  numberOfProperties: number;
+}
+
+interface CategoryData {
+  categories: Category[];
+}
+
+export const fetchCategories = async (): Promise<Category[]> => {
+  const { data } = await client.query<CategoryData>({
+    query: GET_CATEGORIES,
+  });
+  return data.categories;
```

Comment:
Muito boa a implementacao, mas vc já pensou em usar o hook do `useQuery`? 
Ao usar esse hook, vc não precisa lidar com estados, porque o `data` seria um estado e sempre iniciará como `undefined` pelo apollo client. Sendo assim, só teria que retornar todo o objeto de resposta do hook da seguite forma:

```typescript
export const fetchCategories = () => {
  const { data, loading, error } = useQuery<CategoryData>({
    query: GET_CATEGORIES,
  });

  return { data, loading, error };
}
```

Perceba que dessa forma, vc não precisa lidar com o async/await e vc teria os estados de `loading` e de `erro` que, por sua vez, têm seus estados modificados automaticamente pelo useQuery ao realizar a request.

Por fim, voce teria que usar o `ApolloProvider` para conseguir usar esse hook, conforme é demonstrado [nessa sessão da documentacao](https://www.apollographql.com/docs/react/get-started/#step-4-connect-your-client-to-react) (literalmente a mesma coisa - só mudaria o valor da url).

A implementação pode ser considerada simples, mas se não entendeu alguma coisa, poderia dar uma olhada no que o Wanderson fez no [PR dele](https://github.com/wandersoncbjr/desafio-taqtile/pull/22) ou então me chamar no slack para mais detalhes. 😉 

Caso voce opte por manter usando a assincronissidade, pode ser que o objeto `data` seja undefined por algum erro ou não tenha o campo `categories`, entao pode ser que esse retorno resulte num erro de `undefined is not an object`. Sendo assim, é necessário que vc coloque um optional chaining operator:

```suggestion
  return data?.categories;
```